torch/csrc/autograd/utils/lambda_post_hook.h

```diff
@@ -9,24 +9,36 @@ namespace utils {
 // Turns lambda into a torch::autograd::FunctionPostHook.
 class LambdaPostHook : public torch::autograd::FunctionPostHook {
   using variable_list = std::vector<torch::autograd::Variable>;
+  using fn_type =
+      std::function<variable_list(const variable_list&, const variable_list&)>;
+  using compiled_fn_type = std::function<void(CompiledNodeArgs&)>;
 
  public:
   // The lambda function takes as arguments the outputs and inputs of the
   // autograd function and can modify the outputs of the autograd function by
   // returning a new output if needed.
-  /* implicit */ LambdaPostHook(
-      std::function<variable_list(const variable_list&, const variable_list&)>
-          fn)
-      : fn_(std::move(fn)) {}
+  /* implicit */ LambdaPostHook(fn_type fn) : fn_(std::move(fn)) {}
+
+  LambdaPostHook(fn_type fn, compiled_fn_type compiled_fn)
+      : fn_(std::move(fn)), compiled_fn_(std::move(compiled_fn)) {}
 
   variable_list operator()(
       const variable_list& outputs,
       const variable_list& inputs) override {
     return fn_(outputs, inputs);
   }
 
+  void compiled_args(CompiledNodeArgs& args) override {
+    TORCH_CHECK(
+        compiled_fn_ != nullptr,
+        "LambdaPostHook compiled_fn_ is not defined. Either disable "
+        "compiled_autograd or define compiled_fn_ for the LambdaPostHook.");
+    compiled_fn_(args);
+  }
```

Comment:
This was added because the hope of tracing `LambdaPostHook` to enable eager DDP + compiled_autograd. But since we are not going to do that, we can just make it an no op as you suggested.